multinet/api.py

```diff
@@ -1,48 +1,43 @@
 """Flask blueprint for Multinet REST API."""
-from copy import deepcopy
-from dataclasses import asdict
 from flasgger import swag_from
 from flask import Blueprint, request
 from webargs import fields
 from webargs.flaskparser import use_kwargs
 
-from typing import Any, Optional, List, Dict, cast
+from typing import Any, Optional
 from multinet.types import EdgeDirection, TableType
 from multinet.auth.util import (
     require_login,
     require_reader,
-    is_reader,
     require_writer,
     require_maintainer,
     require_owner,
 )
-from multinet.validation import ValidationFailure, UndefinedKeys, UndefinedTable
-from multinet.types import WorkspacePermissions
 
-from multinet import db, util
+from multinet import util
 from multinet.errors import (
-    ValidationFailed,
     BadQueryArgument,
     MalformedRequestBody,
     AlreadyExists,
     RequiredParamsMissing,
 )
-from multinet.user import current_user, find_user_from_id
 
+from multinet.user import current_user
+from multinet.workspace import Workspace
 
 bp = Blueprint("multinet", __name__)
 
 
 @bp.route("/workspaces", methods=["GET"])
 @swag_from("swagger/workspaces.yaml")
 def get_workspaces() -> Any:
-    """Retrieve list of workspaces."""
+    """Return the list of available workspaces, based on the logged in user."""
     user = current_user()
 
-    # Filter all workspaces based on whether it should be shown to the user who
-    # is logged in.
-    stream = util.stream(w["name"] for w in db.get_workspaces() if is_reader(user, w))
-    return stream
+    if user is not None:
+        return util.stream(user.available_workspaces())
+
+    return util.stream(Workspace.list_public())
```

Comment:
The docstring says "available" but the method call is `list_public()`. Should it be `list_available()`, or are we conflating the meanings of purely public workspaces and all available workspaces?

Reply:
If it gets to this line, it means the user isn't logged in, and thus the only available workspaces are the ones that are public.

Reply:
Ah, my reading comprehension skills failed.

A few comments in this block of code might be in order, to help guide readers as to what's going on.